util/helm/client.go

```diff
@@ -164,6 +167,19 @@ func (c *nativeHelmChart) ExtractChart(chart string, version string, passCredent
 		defer func() { _ = os.RemoveAll(tempDest) }()
 
 		if c.enableOci {
+			// check to see if new password needs to be generated
+			if c.creds.CredsRefreshCmd != "" && c.isExpiredCredentials() {
```

Comment:
although this is how the `argo-image-updater` does credential refresh, I'm not a big fan of arbitrary code execution, and it would also require the `repo-server` to have those binaries available on the filesystem.